File: Intermediate/Topic-001-Introduction-to-Problem-Solving/Homework/Q1-Find-Perfect-Numbers/PerfectNumber.py
def is_perfect(A):
    total = 0
    for i in range(1, A//2 + 1):
        if (A%i == 0):
            total += i

    if total == A:
        return 1
    else:
        return 0



class Solution:
    # @param A : integer
    # @return an integer
    def solve(self, A):
        return is_perfect(A)

    
## Runner code
if __name__ == "__main__":
    number = int(input("Enter the positive integer: "))
    solution = Solution()
    result = solution.solve(number)
    if result == 1:
        print(f"{number} is a perfect number.")
    else:
        print(f"{number} is not a perfect number.")

File: Intermediate/Topic-001-Introduction-to-Problem-Solving/Homework/Q1-Find-Perfect-Numbers/PerfectNumber.c
#include <stdio.h>

int perfect_number(int A) {
     if (A < 1) {
        return 0;
    }

     int idx;
     int sum = 0;
     for (idx = 1; idx <= A/2; idx++) {
         if (A % idx == 0) {
             sum += idx;
         }
     }
     if ( sum == A) {
         return 1;
     } else {
         return 0;
     }
 }
int solve(int A) {
    return perfect_number(A);
}


// Runner code
int main() {
    int number;
    printf("Enter the positive integer: ");
    scanf("%d", &number);

    int prime_status = solve(number);
    if (prime_status) {
        printf("%d is a perfect number.\n", number);
    } else {
        printf("%d is not a perfect number.\n", number);
    }

    return 0;
}